includes/class-wc-payments-account.php

```diff
@@ -330,6 +331,46 @@ public function get_fraud_services_config() {
 		return $filtered_services_config;
 	}
 
+	/**
+	 * Checks if the request is for the Capital view offer redirection page, and redirects to the offer if so.
+	 *
+	 * Only admins are be able to perform this action. The redirect doesn't happen if the request is an AJAX request.
+	 * This method will end execution after the redirect if the user requests and is allowed to view the loan offer.
+	 */
+	public function maybe_redirect_to_capital_offer() {
+		if ( wp_doing_ajax() ) {
+			return;
+		}
+
+		// Safety check to prevent non-admin users to be redirected to the view offer page.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		// This is an automatic redirection page, used to authenticate users that come from the offer email. For this reason
+		// we're not using a nonce. The GET parameter accessed here is just to indicate that we should process the redirection.
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		if ( ! isset( $_GET['wcpay-loan-offer'] ) ) {
+			return;
+		}
+
+		$return_url  = $this->get_overview_page_url();
+		$refresh_url = add_query_arg( [ 'wcpay-loan-offer' => '' ], admin_url( 'admin.php' ) );
+
+		try {
+			$capital_link = $this->payments_api_client->get_capital_link( 'capital_financing_offer', $return_url, $refresh_url );
+
+			$this->redirect_to( $capital_link['url'] );
+		} catch ( API_Exception $e ) {
+			$error_url = add_query_arg(
```

Comment:
We can add a log here, just in case.

Reply:
Most errors are already logged by [`extract_response_body`](https://github.com/Automattic/woocommerce-payments/blob/e33a0ea28d4e466e8443a9782f7788ad9105ad86/includes/wc-payment-api/class-wc-payments-api-client.php#L1711), with the exception being [failing to encode the request body](https://github.com/Automattic/woocommerce-payments/blob/e33a0ea28d4e466e8443a9782f7788ad9105ad86/includes/wc-payment-api/class-wc-payments-api-client.php#L1603). I was afraid logging it again here would clutter the logs quite a bit, and left the logging to the API client.

I'm definitely open to adding a new call here if you don't see any issues with it.

Reply:
Yes you're right, that seems to log errors already, I was more curious about Stripe errors, like "offer don't exists" or "wrong account ID" etc, but they return back an account link even if the loan doesn't exist, so it won't be a problem not logging it, as we don't have any user entered inputs here. Feel free to skip this :)